File: BasicMaths/CheckPalindrome/CheckPalindrome.py
class CheckPalindrome:
    def isPalindrome(self, number):
        reversedNumber = self.reverse(number)
        print(reversedNumber == number)
    
    def reverse(self, number):
        reversedNumber = 0
        while(number > 0):
            reversedNumber = reversedNumber * 10 + (number % 10)
            number //= 10
        
        return reversedNumber
    
    def run(self):
        n = int(input("Enter the number: "))
        self.isPalindrome(n)

palindrome = CheckPalindrome()
palindrome.run()

File: BasicMaths/CheckPalindrome/CheckPalindrome.ts
class CheckPalindrome {
  isPalindrome(n: number): void {
    let reversedNumber: number = this.reverse(n);
    console.log(reversedNumber == n);
  }

  reverse(n: number): number {
    let reversedNumber: number = 0;
    while (n > 0) {
      reversedNumber = reversedNumber * 10 + (n % 10);
      n = Math.floor(n / 10);
    }
    return reversedNumber;
  }
}

const palindrome = new CheckPalindrome();
palindrome.isPalindrome(1221);
palindrome.isPalindrome(123456);
